### ISSUE TITLE
[issue]: Try to kill the thread as soon as the USB device is plugged out

### ISSUE THREAD

**Issue by Suleman-Elahi** [ventoy/Ventoy#2410]
### Official FAQ

- [X] I have checked the official FAQ.

### Ventoy Version

1.0.91

### What about latest release

Yes. I have tried the latest release, but the bug still exist.

### Try alternative boot mode

Yes. I have tried them, but the bug still exist.

### BIOS Mode

UEFI Mode

### Partition Style

GPT

### Disk Capacity

32

### Disk Manufacturer

Kingston

### Image file checksum (if applicable)

None

### What happened?

I have two flash drives:

1: Kingston: 16 GB --> This is one is old and kind of faulty to, so I ordered a new one which is the 2nd flash drive.
2: Lexar: 32 GB > The new flash drive.

So, I formatted the Kingston Drive and then try re- install Ventoy. I was just hoping that reinstalling will fix the pen drive. Basically, what happens is when I copy anything to it, it disconnects, and Windows gives error message "USB Device not Recognized".

The format progress in Ventoy was taking long time (bcz the flash drive is faulty ...), so, I unplugged the flash drive during the format process and plugged the new flash drive that I just bought as I though the software will close itself. But it didn't. Instead, it formatted my new flash drive!!!!! I tried closing the software forcefully, but it kept giving me a popup dialog saying the "thread is running".

I had some data in the new pen drive which is lost now :{

------------------------------------------------------------------------------------------------------------------------
Bottom line is that Ventoy should keep checking if the original flash drive on which the format/installation process was initiated is still plugged and should kill itself imo instantly when a user plugs out the flash drive during the formatting process!!!

Just giving my opinion and letting you know... not demanding anything.

Thanks.
